packages/lit-html/src/lib/lit-html.ts

```diff
@@ -552,15 +593,25 @@ export class NodePart {
     this.__startNode.parentNode!.insertBefore(node, this.__endNode);
   }
 
+  private __commitDirective(value: DirectiveResult) {
+    const directive = value._$litDirective$;
+    if (this.__directive?.constructor !== directive) {
```

Comment:
This `.constructor` access is a bit funky. Is it possible to store the constructor (which is just `directive` here) as `this.__directive`, and store the new instance as `this.__value`? Then we'd know both the class constructor and instance value at the same time.

Reply:
What's funky about the `.constructor` access?

We want to store the value returned by the directive so we can do change detection on it and not require that directives handle that themselves.